cities/crud.py

```diff
@@ -0,0 +1,77 @@
+from sqlalchemy.exc import NoResultFound
+from sqlalchemy.ext.asyncio import AsyncSession
+
+from sqlalchemy import select, insert, update, delete
+
+from cities import models, schemas
+
+
+async def get_all_cities(db: AsyncSession):
+    query = select(models.City)
+    city_list = await db.execute(query)
+    return [city[0] for city in city_list.fetchall()]
+
+
+async def get_city_by_id(city_id: int, db: AsyncSession):
+    query = select(models.City).where(models.City.id == city_id)
+    result = await db.execute(query)
+
+    try:
+        city = result.scalar_one()
+
+    except NoResultFound:
+        return None
+
+    return city
+
+
+async def create_city(db: AsyncSession, city: schemas.CityCreateSchema):
+    query = insert(models.City).values(
+        name=city.name,
+        additional_info=city.additional_info
+    )
+    result = await db.execute(query)
+    await db.commit()
+    resp = {**city.model_dump(), "id": result.lastrowid}
+    return resp
+
+
+async def update_city(db: AsyncSession, city_id: int, city: schemas.CityUpdateSchema):
+    existing_city = await get_city_by_id(city_id=city_id, db=db)
+
+    if existing_city is None:
+        return None
+
+    query = (
+        update(models.City)
+        .where(models.City.id == city_id)
+        .values(
+            name=city.name,
+            additional_info=city.additional_info
+        )
+        .returning(models.City.id)
+    )
+    result = await db.execute(query)
+
+    await db.commit()
+
+    updated_id = result.scalar_one_or_none()
+
+    resp = {**city.model_dump(), "id": result.inserted_primary_key[0]}
```

Comment:
In the `update_city` function, `result.inserted_primary_key[0]` is used to construct the response. However, this is incorrect for an update operation. You should use `updated_id` obtained from `result.scalar_one_or_none()` to construct the response.